OS/DiskOS/terminal.lua

```diff
@@ -432,4 +477,60 @@ function term.loop() --Enter the while loop of the terminal
   end
 end
 
+--Given an input and a list of commands, return a list of strings that can be
+--used as autocomplete suggestion. For example if passed "c" and {"cls, copy"}
+--as parameters, will return {"ls", "opy"}.
+function term.autocomplete(input, commands)
+
+  result = {}
+  for k, f in ipairs(commands) do
+    if f:sub(1, input:len()) == input and input:len() < f:len() then
+      table.insert(result, f:sub(input:len()+1))
```

Comment:
This line could be optimized:
`table.insert` is known to be slow in Lua, and when used inside of loops, it could impact the performance of LIKO-12 slightly.

I would recommend to replace it with this:
```lua
result[#result + 1] = f:sub(#input + 1)
```

Note that I did also replace `input:len()` with `#input`, because it's prettier, yes, `#` is not only for tables, but also works on strings. 😉